### PR TITLE
feat: add verify command

### PR DESCRIPTION
**Pull request by ramirezfernando**
### Notes
Commit 1:
- Add `/verify` command based on [doc](https://docs.google.com/document/d/1AUv2IV02YQ0cCoBzJoU2Z1vE1HNxDZibkyhZ9dZiMGI/edit#heading=h.91v0szvlgvz2)
- Add Axios to make API calls to GoGo

Commit 2:
- Remove `axios`
- Add dynamic production and development API url by adding `.env.development` and `.env.production` files with `VERIFY_DISCORD_API_ENDPOINT` in both

https://github.com/codecoogs/bot/assets/91701930/b26cac81-cd57-49cd-8855-d708d4dfa2de

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   },
   "dependencies": {
     "@discordjs/rest": "^1.0.1",
+    "axios": "^1.6.2",
```
Comment (bnleft) [inline package.json:8]: **suggestion:** avoid using axios, let's use `fetch` which should be built-in javascript

File: src/commands/verify.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            .setDescription("Your email address")
+            .setRequired(true)
+    ),
+    execute: async ({ interaction })=> {
+        try {
+            await interaction.deferReply({ ephemeral: true });
+
+            const userEmail = interaction.options.get("email")?.value;
+            const discordId = interaction.user.id
+            const verifyUser = await axios.patch(`http://localhost:3000/v1/users/discord/verify?email=${userEmail}&discordId=${discordId}`)
```
Comment (bnleft) [inline src/commands/verify.ts:27]: **refactor:** we should dynamically know when to use the development API url and production API url

File: src/types.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ interface CommandExecuteOptions {
 type CommandExecute = (options: CommandExecuteOptions) => Promise<any> | any;
 
 export type CoCommandOptions = {
-    data: SlashCommandBuilder;
+    data: Partial<SlashCommandBuilder>;
```
Comment (bnleft) [inline src/types.ts:16]: **question:** What does this change do?

**Review by bnleft — CHANGES_REQUESTED** [review]
**suggestion:** attach a demo gif or video in the PR notes

File: package.json (hunk 1 above)
Comment (ramirezfernando) [reply package.json:8]: Why should we avoid using axios?

File: src/types.ts (hunk 3 above)
Comment (bnleft) [reply src/types.ts:16]: I see, looks like `addStringOption` returns a different type - we can keep your change

**Review by bnleft — APPROVED** [review]
thanks for the demo as well

**Pull request merged**